### PR TITLE
new feature: httpOnly config option

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
This does not prevent the HTTPS server from getting created which happens here: https://github.com/digitalbazaar/bedrock-server/blob/master/lib/index.js#L89

But it DOES prevent the default behavior which is an automatic redirect from HTTP to HTTPS.  This allows connections to be successfully made to the HTTP port.

File: lib/index.js [hunk 1]
```diff
@@ -171,10 +171,13 @@ bedrock.events.on('bedrock.start', async () => {
   // ready check
   const cancel = await bedrock.events.emit('bedrock-express.ready', server);
   // check if default attach canceled
-  if(cancel !== false) {
+  if(!bedrock.config.express.httpOnly || cancel !== false) {
```
Comment (dlongley) [inline lib/index.js:174]: I'm confused by this -- which may be related to the original code as well. Is there some kind of double negative going on here? Previously, if `cancel` was anything but `false`, we would attach express to the HTTPS server. I think that was probably backwards given the variable name. My understanding is that a return value of `false` means the event *was* canceled. So perhaps we should do:

```js
const canceled = await bedrock.events.emit('bedrock-express.ready', server) === false;
```

 And then:

```suggestion
  if(!bedrock.config.express.httpOnly && !canceled) {
```

Which would then read more clearly as: If `httpOnly` isn't set and we haven't canceled, attach to HTTPS.

File: lib/index.js [hunk 2]
```diff
@@ -171,10 +171,13 @@ bedrock.events.on('bedrock.start', async () => {
...
     // attach express to TLS
     bedrockServer.servers.https.on('request', server);
   }
+  if(bedrock.config.express.httpOnly) {
+    bedrockServer.servers.http.on('request', server);
+  }
```
Comment (dlongley) [inline lib/index.js:180]: I think that if the event was canceled (see above) we should not be attaching here. So, now reading this bit ... I think in total we want something like this:

```js
// ready check
const canceled = await bedrock.events.emit('bedrock-express.ready', server) === false;
// do not attach if ready was canceled
if(canceled) {
  return;
}
if(bedrock.config.express.httpOnly) {
  // attach express to HTTP only
  bedrockServer.servers.http.on('request', server);
} else {
  // attach express to TLS
  bedrockServer.servers.https.on('request', server);
}
```

Comment (mattcollier) [reply lib/index.js]: Done.

**Comment by mattcollier** [comment]
The next consideration is how we go about releasing this.  We've already pinned packages to use 2.4.1 due to http2 issues.  Do we make a patch release that rolls back the http2 stuff since it's broken?

**Comment by dlongley** [comment]
> The next consideration is how we go about releasing this. We've already pinned packages to use 2.4.1 due to http2 issues. Do we make a patch release that rolls back the http2 stuff since it's broken?

Yes, I'm ok with that approach.

**Pull request merged**